mla/threeml/models.py

```diff
@@ -122,30 +121,35 @@ def _init_sob_ratio(self, *args, **kwargs) -> None:
                                      bins=bins, weights=sig_w, density=True)
 
         # Normalize histograms by dec band
+        with np.errstate(divide='ignore'):  # divide zero warnings
+            sig_h /= np.sum(sig_h, axis=1)[:, None]
 
-        sig_h /= np.sum(sig_h, axis=1)[:, None]
         if 'k' not in kwargs:
             kwargs['k'] = 1
         if 's' not in kwargs:
             kwargs['s'] = 0
         if 'ext' not in kwargs:
             kwargs['ext'] = 3
-        ratio = sig_h / self._background_sob_map
-        for i in range(ratio.shape[0]):
-            # Pick out the values we want to use.
-            # We explicitly want to avoid NaNs and infinities
-            values = ratio[i]
-            good = np.isfinite(values) & (values > 0)
-            x_good, y_good = bin_centers[good], values[good]
-
-            # Do a linear interpolation across the energy range
-            if len(x_good) > 1:
-                spline = Spline(x_good, y_good, *args, **kwargs)
-                ratio[i] = spline(bin_centers)
-            elif len(x_good) == 1:
-                ratio[i] = y_good
-            else:
-                ratio[i] = 0
+
+        with np.errstate(divide='ignore'):  # divide zero warnings
+            ratio = sig_h / self._background_sob_map
+
+        with np.errstate(divide='ignore'):  # divide zero warnings
```

Comment:
is this the right filter you want here? please change if necessary (same goes for other instances)

Reply:
Yes. I think you can merge it now.